docs/migration/v12.md

```diff
@@ -0,0 +1,329 @@
+
+## Getting started
+
+By running these steps, you should be able to run your app right away:
+
+1. `yarn add @dhis2/app-runtime @dhis2/ui -D @dhis2/cli-app-scripts`
+2. `npx yarn-deduplicate yarn.lock && yarn`
+3. Try out `yarn start --allowJsxInJs`, and your app should be running 🚀
+
+There will be some other changes you will probably want to make, which are described in the following sections. Our goal is to make it easy to adopt the new changes, so we have some tools to facilitate the process.
+
+## Vite & JSX
+
+As suggested by the `--allowJsxInJs` flag, Vite does not allow JSX syntax in files without a .jsx or .tsx extension by default; attempting to run `start` or `build` in that case will throw an error like `✘ [ERROR] The JSX syntax extension is not currently enabled`.
+
+JSX in .js files is not supported for performance reasons: it saves needing to parse non-JSX files for JSX syntax. We can add complicated config to enable parsing of JSX in .js files, but that approach has several downsides:
+
+1. It’s nonstandard — it is most correct to use the right file extensions
+2. It requires more config to maintain
+3. It has performance costs on startup and when parsing files
+   The new default for `@dhis2/cli-app-scripts`, therefore, will be the same as Vite’s. To make the transition easier, though, we provide some tools to 1) support JSX in .js files temporarily, and 2) easily migrate JS files to .jsx extensions if they contain JSX syntax.
+
+### Allow JSX in JS temporarily
+
+To make it as fast as possible to upgrade and get up and running with Vite and React 18, we added an `--allowJsxInJs` flag that you can pass to `start` and `build` scripts, so that you can test out the changes without any file renaming. In many cases, you should be able to upgrade your platform dependencies and run `start` with the flag and see your app running! Other times, you may need to add a few tweaks — see the troubleshooting and breaking changes sections below.
+
+Due to the downsides mentioned above, however, we will remove the `--allowJsxInJs` option in the next major version. When you are ready to rename your files, you can use the migration script we’ve made.
+
+### .js to .jsx migration script
+
+We made a codemod that can migrate your codebase to the right file extensions quickly! It checks .js files for JSX syntax, updates the extension if needed, then updates imports within the file tree to target the new filenames.
+
+```sh
+yarn d2-app-scripts migrate js-to-jsx
+```
+
+Check out the [script docs](../scripts/migrate/js-to-jsx.md) for options, tips, and caveats.
+
+## React 18
+
+In terms of the React 18-related changes, upgrading to `@dhis2/cli-app-scripts` version 12 should generally a smooth process for the consuming apps. Most of the [changes](https://react.dev/blog/2022/03/08/react-18-upgrade-guide) shouldn't affect most apps.
```

Comment:
I feel wording could be changed slightly here for readability. Made a suggestion

```suggestion
With this update we also upgraded from React 16 to React 18. Upgrading to `@dhis2/cli-app-scripts` version 12 should generally a smooth process for the consuming apps. Most of the [changes](https://react.dev/blog/2022/03/08/react-18-upgrade-guide) shouldn't affect most apps.
```